plugins/backup/veeam/src/main/java/org/apache/cloudstack/backup/veeam/VeeamClient.java

```diff
@@ -0,0 +1,661 @@
+            final String veeamVmRefId = lookupVM(heirarchyId, vmwareInstanceName);
+            final HttpResponse response = get(String.format("/jobs/%s/includes", jobId));
+            checkResponseOK(response);
+            final ObjectMapper objectMapper = new XmlMapper();
+            objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            final ObjectsInJob jobObjects = objectMapper.readValue(response.getEntity().getContent(), ObjectsInJob.class);
+            if (jobObjects == null || jobObjects.getObjects() == null) {
+                LOG.warn("No objects found in the Veeam job " + jobId);
+                return false;
+            }
+            for (final ObjectInJob jobObject : jobObjects.getObjects()) {
+                if (jobObject.getName().equals(vmwareInstanceName) && jobObject.getHierarchyObjRef().equals(veeamVmRefId)) {
+                    final HttpResponse deleteResponse = delete(String.format("/jobs/%s/includes/%s", jobId, jobObject.getObjectInJobId()));
+                    return checkTaskStatus(deleteResponse);
+                }
+            }
+            LOG.warn(vmwareInstanceName + " VM was not found to be attached to Veaam job (backup offering): " + jobId);
+            return false;
+        } catch (final IOException e) {
+            LOG.error("Failed to list Veeam jobs due to:", e);
+            checkResponseTimeOut(e);
+        }
+        return false;
+    }
+
+    public boolean restoreFullVM(final String vmwareInstanceName, final String restorePointId) {
+        LOG.debug("Trying to restore full VM: " + vmwareInstanceName + " from backup");
+        try {
+            final HttpResponse response = post(String.format("/vmRestorePoints/%s?action=restore", restorePointId), null);
+            return checkTaskStatus(response);
+        } catch (final IOException e) {
+            LOG.error("Failed to restore full VM due to: ", e);
+            checkResponseTimeOut(e);
+        }
+        throw new CloudRuntimeException("Failed to restore full VM from backup");
+    }
+
+    /////////////////////////////////////////////////////////////////
+    //////////////// Public Veeam PS based APIs /////////////////////
+    /////////////////////////////////////////////////////////////////
+
+    /**
+     * Generate a single command to be passed through SSH
+     */
+    protected String transformPowerShellCommandList(List<String> cmds) {
+        StringJoiner joiner = new StringJoiner(";");
+        joiner.add("PowerShell Add-PSSnapin VeeamPSSnapin");
+        for (String cmd : cmds) {
+            joiner.add(cmd);
+        }
+        return joiner.toString();
+    }
+
+    /**
+     * Execute a list of commands in a single call on PowerShell through SSH
+     */
+    protected Pair<Boolean, String> executePowerShellCommands(List<String> cmds) {
+        try {
+            // FIXME: make ssh timeouts configurable
```

Comment:
Here too

Reply:
This again is a systemic issue, I'm using the same options used in other classes for the ssh utility.